### ISSUE TITLE
when open a file, a character 'p' show in the first line

### ISSUE THREAD

**Issue by kaienkira** [vim/vim#3083]
OS: Ubuntu18.04
version: VIM - Vi IMproved 8.0 (2016 Sep 12, compiled Apr 10 2018 21:31:58)
Included patches: 1-1453
Modified by pkg-vim-maintainers@lists.alioth.debian.org
Compiled by pkg-vim-maintainers@lists.alioth.debian.org

software: SecureCRT6.2.3
term: xterm

![1](https://user-images.githubusercontent.com/1057578/41821962-3fd8dddc-781b-11e8-916f-d8ff854abc59.PNG)

any work around?

**Comment by kaienkira** [comment]
i have read the issue #2008 and nothing lucky.

**Comment by tonymec** [comment]
There used to be a patch or two for a very similar problem but I can't find them back.

Before reporting a problem, it is always best to check first if it still happens with the latest source. After Vim 8.0.1453, there has been 511 changes to the source already (8.0.1454 to 8.0.1850 and 8.1.0 to 8.1.113). In addition, Debian maintainers are known for making not always well-advised changes in the sources. Please check if you have the same problem with Bram's unmodified latest sources.

See:
        http://vim.wikia.com/wiki/Getting_the_Vim_source_with_Mercurial
        http://users.skynet.be/antoine.mechelynck/vim/compunix.htm

Best regards,
Tony.

**Comment by vim-ml** [comment]
Tony Mechelynck <vim-dev-github@256bit.org> wrote:

> See:
>         http://vim.wikia.com/wiki/Getting_the_Vim_source_with_Mercurial
>         http://users.skynet.be/antoine.mechelynck/vim/compunix.htm

Above instructions to build Vim are a bit complicated in my
opinion. I find instructions in the following link simpler:

http://vim.wikia.com/wiki/Building_Vim

Dominique

**Comment by jamessan** [comment]
> In addition, Debian maintainers are known for making not always well-advised changes in the sources.

I'd appreciate it if you stopped with the unfounded bad-mouthing.  The mere fact that someone is experiencing a problem with a package provided by someone other than Bram does not mean it is caused by the packager.

Debian's Vim packaging has very few patches and I specifically strive to maintain minimal deviation from upstream Vim.  You can readily inspect the [integrated patches](https://sources.debian.org/patches/vim/) of all the currently released Debian Vim packages.

> software: SecureCRT6.2.3

At a guess, this doesn't understand the escape sequence to request the blink status of the cursor and is printing it to the screen instead of ignoring it (as it should for any unhandled CSI sequence).  The version of SecureCRT is fairly old, so if it is a SecureCRT issue, it may be fixed in a newer version.

If you were to run `printf '\e[?12$p'; cat` at the shell inside of SecureCRT, you would probably see the `$p`.  If so, that would confirm the issue.

It would also be good to know what `:echo v:termresponse` in Vim reports.  That may allow Vim to work around the issue.

**Comment by tonymec** [comment]
OK, OK, let's calm down. I should have said:

Before reporting a problem, it is always best to check first if it still happens with the latest original sources. After Vim 8.0.1453, there has been 511 changes to the source already (8.0.1454 to 8.0.1850 and 8.1.0 to 8.1.113). Please check if you have the same problem with Bram's unmodified latest sources.

and not mentioned Debian at all.

**Comment by k-takata** [comment]
Quoting from Bram's comment:

https://github.com/vim/vim/issues/2008#issuecomment-396543497
> The last fix for term responses was in 8.0.1556. I'm not aware of any remaining problems since then.

So, please use 8.0.1556 or later, and also use a correct setting for $TERM.

**Comment by ttdoda** [comment]
I think #2126 fixes this problem, and it fixes old iTerm2's problem too.
https://github.com/vim/vim/issues/2008#issuecomment-396562161

**Comment by k-takata** [comment]
@ttdoda Do you mean that 8.0.1556 is not enough for this SecureCRT problem?

I thought that the advantage of #2126 to 8.0.1556 was that it worked correctly even if a user set $TERM to "xterm" wrongly (e.g. using old iTerm2).

**Comment by kaienkira** [comment]
I have tried to compile the lastest release, but no lucky

VIM - Vi IMproved 8.1 (2018 May 18, compiled Jun 30 2018 09:31:11)
Included patches: 1-127

![default](https://user-images.githubusercontent.com/1057578/42120544-8a376352-7c4f-11e8-8e90-1d58f6e73a70.png)

:echo v:termrespons return
^[[>1;10;0c

**Comment by tonymec** [comment]
According to `:help v:termresponse`, the second number (here 10) should always be 95 or bigger: so obviously you aren't running in a true xterm-compatible terminal. As a matter of comparison, I see here `^[[>0;115;0c` in konsole, `^[[>41;330;0c` in xterm and `^[[>24;279;0c` in mlterm.

What kind of terminal are you running in and what does it set $TERM to?

Best regards,
Tony.

**Comment by jamessan** [comment]
This will likely help weed out non-compliant terminals:

```diff
diff --git i/src/term.c w/src/term.c
index 4bb8030cb..29c1d3c97 100644
--- i/src/term.c
+++ w/src/term.c
@@ -4690,6 +4690,11 @@ check_termcode(
 				&& STRNCMP(tp + extra - 2, "0;115;0c", 8) == 0)
 			    is_not_xterm = TRUE;
 
+			/* xterm first responded to this request at patch level
+			 * 95, so assume anything below 95 is not xterm */
+			if (version < 95)
+			    is_not_xterm = TRUE;
+
 			/* Only request the cursor style if t_SH and t_RS are
 			 * set. Only supported properly by xterm since version
 			 * 279 (otherwise it returns 0x18).
```

> What kind of terminal are you running in and what does it set $TERM to?

As mentioned in the original post:

software: SecureCRT6.2.3
term: xterm

**Comment by brammool** [comment]
James McCoy wrote:

> This will likely help weed out non-compliant terminals:
> 
> ```diff
> diff --git i/src/term.c w/src/term.c
> index 4bb8030cb..29c1d3c97 100644
> --- i/src/term.c
> +++ w/src/term.c
> @@ -4690,6 +4690,11 @@ check_termcode(
>  				&& STRNCMP(tp + extra - 2, "0;115;0c", 8) == 0)
>  			    is_not_xterm = TRUE;
>  
> +			/* xterm first responded to this request at patch level
> +			 * 95, so assume anything below 95 is not xterm */
> +			if (version < 95)
> +			    is_not_xterm = TRUE;
> +
>  			/* Only request the cursor style if t_SH and t_RS are
>  			 * set. Only supported properly by xterm since version
>  			 * 279 (otherwise it returns 0x18).
> ```
> 
> > What kind of terminal are you running in and what does it set $TERM to?
> 
> As mentioned in the original post:
> 
> software: SecureCRT6.2.3
> term: xterm

I like the idea.  Xterm 95 is ancient anyway.

-- 
We are the Borg of GNU GPL.  We will assimilate your source code.
Resistance is futile.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by k-takata** [comment]
I think this is fixed by 8.1.0129: https://github.com/vim/vim/commit/668324ef4fd59980dc57b2eefe57eddc57c7b920.
Closing.

**Comment by kaienkira** [comment]
Confirmed fixed
![default](https://user-images.githubusercontent.com/1057578/42158319-acfc661a-7e22-11e8-8f0d-ba3a95b4d7b4.png)
